SimCalorimetry/EcalSelectiveReadoutProducers/plugins/EcalSRCondTools.cc

```diff
@@ -0,0 +1,150 @@
+    edm::ESGetToken<EcalTPGPhysicsConst, EcalTPGPhysicsConstRcd> tpgPhysicsConstToken_;
+  };
+
+  using namespace std;
+  using namespace ecalsrcondtools;
+
+  EcalSRCondTools::EcalSRCondTools(const edm::ParameterSet& ps)
+      : ps_(ps), mode_(ps.getParameter<string>("mode")), iomode_write_(true), done_(false) {
+    if (mode_ == "read") {
+      iomode_write_ = false;
+      hSrToken_ = esConsumes();
+      tpgPhysicsConstToken_ = esConsumes();
+    }
+  }
+
+  EcalSRCondTools::~EcalSRCondTools() {}
+
+  void EcalSRCondTools::analyze(const edm::Event& event, const edm::EventSetup& es) {
+    if (done_)
+      return;
+    EcalSRSettings sr;
+
+    if (mode_ == "online_config" || mode_ == "combine_config") {
+      string fname = ps_.getParameter<string>("onlineSrpConfigFile");
+      ifstream f(fname.c_str());
+      if (!f.good()) {
+        throw cms::Exception("EcalSRCondTools") << "Failed to open file " << fname;
+      }
+      importSrpConfigFile(sr, f, true);
+    }
+
+    if (mode_ == "python_config" || mode_ == "combine_config") {
+      importParameterSet(sr, ps_);
+    }
+
+    if (!(mode_ == "python_config" || mode_ == "online_config" || mode_ == "combine_config" || (mode_ == "read"))) {
+      throw cms::Exception("Config") << "Invalid value," << mode_ << ",  for parameter mode. "
+                                     << "Valid values: online_config, python_config, combine_config, read";
+    }
+
+    if (iomode_write_) {
+      sr.bxGlobalOffset_ = ps_.getParameter<int>("bxGlobalOffset");
+      sr.automaticSrpSelect_ = ps_.getParameter<int>("automaticSrpSelect");
+      sr.automaticMasks_ = ps_.getParameter<int>("automaticMasks");
+
+      edm::Service<cond::service::PoolDBOutputService> db;
+      if (!db.isAvailable()) {
+        throw cms::Exception("CondDBAccess") << "Failed to connect to PoolDBOutputService\n";
+      }
+      //fillup DB
+      //create new infinite IOV
+      cond::Time_t firstSinceTime = db->beginOfTime();
+      db->writeOneIOV(sr, firstSinceTime, "EcalSRSettingsRcd");
+      done_ = true;
+    } else {  //read mode
+      const edm::ESHandle<EcalSRSettings> hSr = es.getHandle(hSrToken_);
+      if (!hSr.isValid()) {
+        cout << "EcalSRSettings record not found. Check the Cond DB Global tag.\n";
+      } else {
+        const EcalSRSettings* ssr = hSr.product();
+        cout << "ECAL Seletive readout settings:\n";
+        cout << *ssr << "\n" << endl;
+      }
+
+      //trigger tower thresholds (from FENIX configuration):
+      const edm::ESHandle<EcalTPGPhysicsConst> hTp = es.getHandle(tpgPhysicsConstToken_);
+      if (!hTp.isValid()) {
+        cout << "EcalTPGPhysicsConst record not found. Check the Cond DB Global tag.\n";
+      } else {
+        const EcalTPGPhysicsConst* tp = hTp.product();
+        const EcalTPGPhysicsConstMap& mymap = tp->getMap();
+        if (mymap.size() != 2) {
+          cout << "Error: TPG physics record is of unexpected size: " << mymap.size()
```

Comment:
@Dr15Jones , is it OK to left cout in this class. Does it mean that it is allowed for "one" producer? My problem with these cout - it is not obvious which class is printing
.

Reply:
#41789  is created , so this PR may be merged.